### PR TITLE
feat: make peeraddress case-safe

### PR DESCRIPTION
**Pull request by elisealix22**
Resolves https://github.com/xmtp/xmtp-js/issues/172

This PR updates `Client#getUserContact` to be case-safe so we can use `canMessage()` and `newConversation()` passing a lowercase address for example.

### CODE REVIEW

File: src/Client.ts [hunk 1]
```diff
@@ -213,23 +213,23 @@ export default class Client {
    * Returns the cached PublicKeyBundle if one is known for the given address or fetches
    * one from the network
    */
-
   async getUserContact(
     peerAddress: string
   ): Promise<PublicKeyBundle | undefined> {
-    const existingBundle = this.knownPublicKeyBundles.get(peerAddress)
+    const safeAddress = ethers.utils.getAddress(peerAddress)
```
Comment (neekolas) [inline src/Client.ts:219]: Maybe we want to move this to `utils.ts`

**Review by neekolas — APPROVED** [review]

**Comment by elisealix22** [comment]
This is kinda frozen right now as we didn't end up needing it for Lenster and one of the test failures is legitimate in that if ethers errors, I will have to provide a fallback. Going to keep it open to revisit next week and handle that edge case / fix tests.

**Comment by elisealix22** [comment]
Closing because it just makes more sense to start fresh on 7.0 on my next, more-focused attempt at this.
